packages/core/src/baseclient.ts

```diff
@@ -139,6 +153,17 @@ export abstract class BaseClient<B extends Backend, O extends Options> implement
   public captureEvent(event: Event, hint?: EventHint, scope?: Scope): string | undefined {
     let eventId: string | undefined = hint && hint.event_id;
 
+    if (this._options.autoSessionTracking) {
+      const isTransaction = event.type === 'transaction';
+
+      if (!isTransaction && scope) {
```

Comment:
Negative filtering here seems a bit dangerous, what about all other event types, present and future, do they cause a session to be reported?!

What I'm actually missing here is checking whether the event contains an exception or not.

Reply:
I see your point, but I did it this way because after talking with @kamilogorek , he told me that we only have two event types atm which are `exception` and `transaction`, and for `exception`, we do not assign a type.
So following that logic, at the moment, to check if it is an exception or not, I need to negatively check if event type is a `transaction`

but I agree, it is not the best way to do it